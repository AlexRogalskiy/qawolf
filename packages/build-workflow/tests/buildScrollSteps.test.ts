import { loadEvents } from "@qawolf/fixtures";
import { buildScrollSteps } from "../";

describe("buildScrollSteps", () => {
  let events: any[];

  beforeAll(async () => {
    events = await loadEvents("scroll_login");
  });

  it("builds expected steps", async () => {
    const steps = buildScrollSteps(events);
    expect(steps).toMatchSnapshot();
  });
});
