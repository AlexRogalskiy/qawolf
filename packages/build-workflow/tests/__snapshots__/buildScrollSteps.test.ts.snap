// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildScrollSteps builds expected steps 1`] = `
Array [
  Object {
    "action": "scroll",
    "html": Object {
      "ancestors": Array [],
      "node": Object {
        "attrs": Object {},
        "children": Array [],
        "name": "html",
        "type": "tag",
        "voidElement": true,
      },
    },
    "index": 25,
    "isFinal": true,
    "page": 0,
    "value": Object {
      "x": 0,
      "y": 378,
    },
  },
]
`;
